srunner/scenarios/parking_exit.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2018-2020 Intel Corporation
+#
+# This work is licensed under the terms of the MIT license.
+# For a copy, see <https://opensource.org/licenses/MIT>.
+
+"""
+Scenario in which the ego is parked between two vehicles and has to maneuver to start the route.
+"""
+
+from __future__ import print_function
+
+import py_trees
+import carla
+
+
+from srunner.scenariomanager.carla_data_provider import CarlaDataProvider
+from srunner.scenariomanager.scenarioatomics.atomic_behaviors import SwitchOutsideRouteLanesTest, ActorSource, ActorTransformSetter
+from srunner.scenariomanager.scenarioatomics.atomic_criteria import CollisionTest
+from srunner.scenariomanager.scenarioatomics.atomic_trigger_conditions import InTriggerDistanceToLocation
+from srunner.scenarios.basic_scenario import BasicScenario
+
+import math
+import operator
+
+
+def convert_dict_to_location(actor_dict):
+    """
+    Convert a JSON string to a Carla.Location
+    """
+    location = carla.Location(
+        x=float(actor_dict['x']),
+        y=float(actor_dict['y']),
+        z=float(actor_dict['z'])
+    )
+    return location
+
+
+def get_same_dir_transform_by_distance(transform, distance: float):
+    """
+    Get a transform in the same direction as the given one.
+    With {distance} (m) offset.
+    Works for arbitrary point on the map.
+    """
+    new_t = carla.Transform()
+    yaw = transform.rotation.yaw
+    new_t.location.x = transform.location.x + \
+        distance*math.cos(math.radians(yaw))
+    new_t.location.y = transform.location.y + \
+        distance*math.sin(math.radians(yaw))
+    new_t.location.z = transform.location.z
+    new_t.rotation = transform.rotation
+    return new_t
+
+
+class ParkingExit(BasicScenario):
+    """
+    This class holds everything required for a scenario in which the ego would be teleported to the parking lane.
+    Once the scenario is triggered, the OutsideRouteLanesTest will be deactivated since the ego is out of the driving lane.
+    Then blocking vehicles will be generated in front of and behind the parking point.
+    The ego need to exit from the parking lane and then merge into the driving lane.
+    After the ego is {end_distance} meters away from the parking point, the OutsideRouteLanesTest will be activated and the scenario ends.
+    
+    Note 1: For route mode, this shall be the first scenario of the route. The trigger point shall be the first point of the route waypoints.
+    
+    Note 2: Make sure there are enough space for spawning blocking vehicles.
+    """
+
+    def __init__(self, world, ego_vehicles, config, debug_mode=False, criteria_enable=True,
+                 timeout=180):
+        """
+        Setup all relevant parameters and create scenario
+        and instantiate scenario manager
+        """
+        self._world = world
+        self._map = CarlaDataProvider.get_map()
+        self.timeout = timeout
+        self._parking_point = convert_dict_to_location(
```

Comment:
Not sure this is needed. We can automatise this by getting the side lane. Or add a parameter like "left" or "right" instead, which is much user friendly than a hardcoded position

Reply:
Added parking_lane_side=left/right parameter